scheduler/replica_scheduler.go

```diff
@@ -559,6 +569,11 @@ func (rcs *ReplicaScheduler) isFailedReplicaReusable(r *longhorn.Replica, v *lon
 		return false
 	}
 
+	tolerateEmptyDiskSelectorVolume, err := rcs.ds.GetSettingAsBool(types.SettingNameTolerateEmptyDiskSelectorVolume)
+	if err != nil {
+		logrus.Errorf("Failed to get tolerate empty tag volume setting: %v", err)
```

Comment:
Should we change `isFailedReplicaReusable` to return bool and error instead? so here, if encountering an error when getting the setting, then we can just return the error back, and no need to print log here.

Or this is intended to ignore the error? if yes, why?

Reply:
originally, I plan to just log error and use the default `false` returned from GetSettingAsBool  like what we did before for `zoneSoftAntiAffinity` and `nodeSoftAntiAffinity` above
```
	zoneSoftAntiAffinity, err := rcs.ds.GetSettingAsBool(types.SettingNameReplicaZoneSoftAntiAffinity)
	if err != nil {
		logrus.Errorf("Error getting replica zone soft anti-affinity setting: %v", err)
        }
```
but you are right, we should return error so we don't get confused behavior when setting is not found
change `zoneSoftAntiAffinity` and `nodeSoftAntiAffinity`  as well

Reply:
Fixed.